rust/datafusion/src/optimizer/filter_push_down.rs

```diff
@@ -93,7 +93,9 @@ fn issue_filters(
     let predicate = predicates
         .iter()
         .skip(1)
-        .fold(predicates[0].clone(), |acc, predicate| and(&acc, predicate));
+        .fold(predicates[0].clone(), |acc, predicate| {
+            and(acc, (*predicate).clone())
```

Comment:
do you think it is worth adding a test to this? Looks non-trivial :/

Reply:
This is just cloning rather than using a reference when creating the `and` expression. `predicate` is `&&Expr` so I dereference with `*` to get `&Expr` and then clone (perhaps could be `to_owned` instead).